python_modules/dagster/dagster/_core/definitions/asset_condition/scheduling_condition_evaluation_context.py

```diff
@@ -0,0 +1,114 @@
+import dataclasses
+import datetime
+import functools
+import logging
+from dataclasses import dataclass
+from typing import AbstractSet, Mapping, Optional, Tuple
+
+import pendulum
+
+from dagster._core.asset_graph_view.asset_graph_view import (
+    AssetGraphView,
+    AssetSlice,
+)
+from dagster._core.definitions.asset_condition.asset_condition import (
+    AssetCondition,
+    AssetConditionEvaluation,
+    AssetConditionEvaluationState,
+)
+from dagster._core.definitions.asset_key import AssetKey
+from dagster._core.definitions.asset_subset import ValidAssetSubset
+from dagster._core.definitions.events import AssetKeyPartitionKey
+
+from .asset_condition_evaluation_context import AssetConditionEvaluationContext
+
+
+@dataclass(frozen=True)
+class SchedulingConditionEvaluationContext:
+    # the AssetKey of the currently-evaluated asset
+    asset_key: AssetKey
```

Comment:
would like to do a bit of thinking about the asset check key case here

Reply:
Just checking back in here. @OwenKephart I merged this entire stack [here](https://github.com/dagster-io/dagster/pull/21582) and looked through it and I really like the direction it is heading. So I am largely into this stack going skip pushing back stuff that I know will nor survive the stack, even if I would object to it normally :-)